### PR TITLE
added specs for pivot facet work in facets_helper #274

### LINKED ISSUE #274: Specs for facets_helper.rb

### PR DESCRIPTION
**Pull request by banukutlu**
Closes #274

### CODE REVIEW

File: spec/helpers/facets_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe FacetsHelper do
+  let(:basic_facet) { Blacklight::Solr::Response::Facets::FacetItem.new(value: 'some value', field: 'basic_facet') }
+  let(:child_item) do
+    Blacklight::Solr::Response::Facets::FacetItem.new(value: 'child value', field: 'child_item', items: [])
+  end
+  let(:pivot_items) do
```
[the remark this answers is not in the document]
Comment (banukutlu) [reply spec/helpers/facets_helper_spec.rb]: > did you see my comments in the spec itself? especially can we rename pivot_items?

sorry I missed these, will do!

File: spec/helpers/facets_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      end
+    end
+
+    before do
+      without_partial_double_verification do
+        allow(helper).to receive(:blacklight_config).and_return blacklight_config
+      end
+    end
+
+    it 'checks if the facet field is selected in the user params' do
```
Comment (cdmo) [inline spec/helpers/facets_helper_spec.rb]: replace user with request?

File: spec/helpers/facets_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      allow(helper).to receive_messages(params: { f: { 'basic-field' => ['x'] } })
+      expect(helper).to be_facet_field_in_params('basic-field')
+      expect(helper.facet_field_in_params?('other-field')).not_to be true
+    end
+
+    it 'checks if the pivot facet field is selected in the user params' do
```
Comment (cdmo) [inline spec/helpers/facets_helper_spec.rb]: replace user with request?

File: spec/helpers/facets_helper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    before do
+      without_partial_double_verification do
+        allow(helper).to receive(:blacklight_config).and_return blacklight_config
+      end
+    end
+
+    it 'checks if the facet field is selected in the user params' do
+      allow(helper).to receive_messages(params: { f: { 'basic-field' => ['x'] } })
+      expect(helper).to be_facet_field_in_params('basic-field')
+      expect(helper.facet_field_in_params?('other-field')).not_to be true
```
Comment (cdmo) [inline spec/helpers/facets_helper_spec.rb]: could you change this to be consistent with the above line:

`      expect(helper).not_to be_facet_field_in_params('other-field')`

this goes for line 113, 118 and 123.

**Pull request merged**